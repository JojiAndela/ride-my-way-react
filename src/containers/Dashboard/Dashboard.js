import React, { Component } from 'react';

import SideBar from 'Components/SideBar/SideBar';
import MainContent from 'Components/DashboardContent/Dash';

import classes from './Dashboard.scss';

class Dashboard extends Component {
  render() {
    return (
      <div className={classes.Dashboard}>
        <SideBar />
        <MainContent />
      </div>
    );
  }
}

export default Dashboard;
