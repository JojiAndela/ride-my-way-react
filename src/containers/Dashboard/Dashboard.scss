@import '../../assets/styles/_settings.scss';

.Dashboard{
  display: flex;
}
