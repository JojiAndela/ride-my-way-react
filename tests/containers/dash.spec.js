import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';

// Component to be tested
import Dashboard from '../../src/containers/Dashboard/Dashboard';

const initialState ={
}


const middleware = [thunk];
const mockStore = configureStore(middleware);
let wrapper;
let store;
describe('Dashboard', () => {
  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <Dashboard />
        </BrowserRouter>
      </Provider>
    );
  });
  it('should render the Dashboard page correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
