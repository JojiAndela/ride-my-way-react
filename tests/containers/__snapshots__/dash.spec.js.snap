// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should render the Dashboard page correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Dashboard>
        <div
          className="Dashboard"
        >
          <sideBar>
            <div
              className="SideBar"
            >
              <div
                className="inner"
              >
                <header>
                  <div
                    className="header"
                  >
                    <h1>
                       RIDE MY WAY
                    </h1>
                  </div>
                </header>
                <Connect(withRouter(sideNav))>
                  <withRouter(sideNav)
                    signoutAction={[Function]}
                  >
                    <Route>
                      <sideNav
                        history={
                          Object {
                            "action": "POP",
                            "block": [Function],
                            "createHref": [Function],
                            "go": [Function],
                            "goBack": [Function],
                            "goForward": [Function],
                            "length": 1,
                            "listen": [Function],
                            "location": Object {
                              "hash": "",
                              "pathname": "/",
                              "search": "",
                              "state": undefined,
                            },
                            "push": [Function],
                            "replace": [Function],
                          }
                        }
                        location={
                          Object {
                            "hash": "",
                            "pathname": "/",
                            "search": "",
                            "state": undefined,
                          }
                        }
                        match={
                          Object {
                            "isExact": true,
                            "params": Object {},
                            "path": "/",
                            "url": "/",
                          }
                        }
                        signoutAction={[Function]}
                      >
                        <div
                          className="sideNav"
                        >
                          <div
                            className="upper"
                          >
                            <NavLink
                              activeClassName="active"
                              activeStyle={
                                Object {
                                  "borderLeft": "4px teal solid",
                                  "fontWeight": "bold",
                                }
                              }
                              aria-current="page"
                              exact={true}
                              to="/dashboard"
                            >
                              <Route
                                exact={true}
                                path="\\\\/dashboard"
                              >
                                <Link
                                  aria-current={null}
                                  replace={false}
                                  to="/dashboard"
                                >
                                  <a
                                    aria-current={null}
                                    href="/dashboard"
                                    onClick={[Function]}
                                  >
                                    Dashboard
                                  </a>
                                </Link>
                              </Route>
                            </NavLink>
                            <NavLink
                              activeClassName="active"
                              activeStyle={
                                Object {
                                  "borderLeft": "4px teal solid",
                                  "fontWeight": "bold",
                                }
                              }
                              aria-current="page"
                              to="/dashboard/rides"
                            >
                              <Route
                                path="\\\\/dashboard\\\\/rides"
                              >
                                <Link
                                  aria-current={null}
                                  replace={false}
                                  to="/dashboard/rides"
                                >
                                  <a
                                    aria-current={null}
                                    href="/dashboard/rides"
                                    onClick={[Function]}
                                  >
                                    All rides
                                  </a>
                                </Link>
                              </Route>
                            </NavLink>
                            <NavLink
                              activeClassName="active"
                              activeStyle={
                                Object {
                                  "borderLeft": "4px teal solid",
                                  "fontWeight": "bold",
                                }
                              }
                              aria-current="page"
                              to="/dashboard/requests"
                            >
                              <Route
                                path="\\\\/dashboard\\\\/requests"
                              >
                                <Link
                                  aria-current={null}
                                  replace={false}
                                  to="/dashboard/requests"
                                >
                                  <a
                                    aria-current={null}
                                    href="/dashboard/requests"
                                    onClick={[Function]}
                                  >
                                    All requests
                                  </a>
                                </Link>
                              </Route>
                            </NavLink>
                          </div>
                          <a
                            className="logout"
                            href="#"
                            onClick={[Function]}
                          >
                            Sign Out
                          </a>
                        </div>
                      </sideNav>
                    </Route>
                  </withRouter(sideNav)>
                </Connect(withRouter(sideNav))>
              </div>
            </div>
          </sideBar>
          <dashboardContent>
            <div
              className="dashContent"
            >
              <Navbar>
                <div
                  className="Nav"
                >
                  <nav
                    className="NavItems"
                  >
                    <NavLink
                      activeClassName="active"
                      aria-current="page"
                      to="/dashboard/rides/new"
                    >
                      <Route
                        path="\\\\/dashboard\\\\/rides\\\\/new"
                      >
                        <Link
                          aria-current={null}
                          replace={false}
                          to="/dashboard/rides/new"
                        >
                          <a
                            aria-current={null}
                            href="/dashboard/rides/new"
                            onClick={[Function]}
                          >
                            Create New Ride
                          </a>
                        </Link>
                      </Route>
                    </NavLink>
                    <NavLink
                      activeClassName="active"
                      aria-current="page"
                      to="/dashboard"
                    >
                      <Route
                        path="\\\\/dashboard"
                      >
                        <Link
                          aria-current={null}
                          replace={false}
                          to="/dashboard"
                        >
                          <a
                            aria-current={null}
                            href="/dashboard"
                            onClick={[Function]}
                          >
                            Dashboard
                          </a>
                        </Link>
                      </Route>
                    </NavLink>
                  </nav>
                </div>
              </Navbar>
              <Switch>
                <Route
                  component={[Function]}
                  computedMatch={
                    Object {
                      "isExact": true,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    }
                  }
                >
                  <notFound
                    history={
                      Object {
                        "action": "POP",
                        "block": [Function],
                        "createHref": [Function],
                        "go": [Function],
                        "goBack": [Function],
                        "goForward": [Function],
                        "length": 1,
                        "listen": [Function],
                        "location": Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        },
                        "push": [Function],
                        "replace": [Function],
                      }
                    }
                    location={
                      Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      }
                    }
                    match={
                      Object {
                        "isExact": true,
                        "params": Object {},
                        "path": "/",
                        "url": "/",
                      }
                    }
                  >
                    <div
                      className="body"
                    >
                      <svg
                        className="svg"
                        height="500px"
                        version="1.1"
                        viewBox="0 0 837 1045"
                        width="380px"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          className="Page1"
                          fill="none"
                          fillRule="evenodd"
                          stroke="none"
                          strokeWidth="1"
                        >
                          <path
                            className="Polygon1"
                            d="M353,9 L626.664028,170 L626.664028,487 L353,642 L79.3359724,487 L79.3359724,170 L353,9 Z"
                            stroke="#007FB2"
                            strokeWidth="6"
                          />
                          <path
                            className="Polygon2"
                            d="M78.5,529 L147,569.186414 L147,648.311216 L78.5,687 L10,648.311216 L10,569.186414 L78.5,529 Z"
                            stroke="#EF4A5B"
                            strokeWidth="6"
                          />
                          <path
                            className="Polygon3"
                            d="M773,186 L827,217.538705 L827,279.636651 L773,310 L719,279.636651 L719,217.538705 L773,186 Z"
                            stroke="#795D9C"
                            strokeWidth="6"
                          />
                          <path
                            className="Polygon4"
                            d="M639,529 L773,607.846761 L773,763.091627 L639,839 L505,763.091627 L505,607.846761 L639,529 Z"
                            stroke="#F2773F"
                            strokeWidth="6"
                          />
                          <path
                            className="Polygon5"
                            d="M281,801 L383,861.025276 L383,979.21169 L281,1037 L179,979.21169 L179,861.025276 L281,801 Z"
                            stroke="#36B455"
                            strokeWidth="6"
                          />
                        </g>
                      </svg>
                      <div
                        className="messageBox"
                      >
                        <h1>
                          404
                        </h1>
                        <p>
                          Page not found
                        </p>
                        <div
                          className="buttonsCon"
                        >
                          <div
                            className="actionLinkWrap"
                          >
                            <Link
                              className="linkButton"
                              replace={false}
                              to="/"
                            >
                              <a
                                className="linkButton"
                                href="/"
                                onClick={[Function]}
                              >
                                Go to Home Page
                              </a>
                            </Link>
                          </div>
                        </div>
                      </div>
                    </div>
                  </notFound>
                </Route>
              </Switch>
            </div>
          </dashboardContent>
        </div>
      </Dashboard>
    </Router>
  </BrowserRouter>
</Provider>
`;
